cpp/inc/bond/ext/grpc/server.h

```diff
@@ -56,17 +56,51 @@
 #include <boost/optional/optional.hpp>
 
 #include <memory>
+#include <string>
 #include <thread>
 
 namespace bond { namespace ext { namespace gRPC
 {
     /// @brief Models a gRPC server powered by Bond services.
     ///
-    /// Servers are configured and started via
-    /// bond::ext:gRPC::server_builder.
+    /// Servers are configured and started via bond::ext:gRPC::server::Start.
     class server final
     {
     public:
+        template <typename... Services>
+        static server Start(
+            grpc::ServerBuilder& builder,
+            std::unique_ptr<Services>... services)
+        {
+            std::vector<std::unique_ptr<detail::service>> all;
+            std::initializer_list<int>{ (all.emplace_back(std::move(services)), 0)... };
+            return Start(builder, std::move(all));
+        }
+
+        template <typename... Services>
+        static server Start(
+            grpc::ServerBuilder& builder,
+            std::pair<std::string, std::unique_ptr<Services>>... namedServices)
+        {
+            std::vector<std::pair<std::string, std::unique_ptr<detail::service>>> all;
+            std::initializer_list<int>{ (all.emplace_back(std::move(namedServices)), 0)... };
+            return Start(builder, std::move(all));
+        }
+
+        static server Start(
+            grpc::ServerBuilder& builder,
+            std::vector<std::unique_ptr<detail::service>> services)
```

Comment:
If a client of `server::Start` wants to have a vector of heterogeneous services, they have to mention a detail type when creating that vector. Should we extract out the parts that `server` needs into an abstract, non-detail interface?

I think we just need:
* `start`
* `grpc_service`
* `SetCompletionQueue`

This is an odd, clunky interface. Any better suggestions?

Reply:
Added an empty `bond::ext::gRPC::abstract_service`.